common/scala/src/main/scala/org/apache/openwhisk/common/NestedSemaphore.scala

```diff
@@ -100,14 +100,19 @@ class NestedSemaphore[T](memoryPermits: Int) extends ForcibleSemaphore(memoryPer
     if (maxConcurrent == 1) {
       super.release(memoryPermits)
     } else {
-      val concurrentSlots = actionConcurrentSlotsMap(actionid)
-      val (memoryRelease, actionRelease) = concurrentSlots.release(1, true)
-      //concurrent slots
-      if (memoryRelease) {
-        super.release(memoryPermits)
-      }
-      if (actionRelease) {
-        actionConcurrentSlotsMap.remove(actionid)
+      //This map may be recreated (multiple times) due to cluster membership events, so don't assume the entry exists...
+      actionConcurrentSlotsMap.get(actionid) match {
```

Comment:
```suggestion
      actionConcurrentSlotsMap.get(actionid).foreach { concurrentSlots =>
```

Would spare you from having the awkward empty `None` case

Reply:
👍

Reply:
The stack trace is added to logging here for the catastrophic case where message consumer processing fails - previously there is just a stack trace added without indicating it was part of message processing etc. This is slightly better, I think, since it provides an ERROR log, and also associates it with message consumption.